### PR TITLE
[FEATURE] Add support for Interpolation Method for Quantile Statistic Used by Estimators in NumericMetricRangeMultiBatchParameterBuilder

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Utilize the `method` setting in `np.quantile()`, which is used by both `bootstrap` and `oneshot` estimation methods (for `NumericMetricRangeMultiBatchParameterBuilder`).  If initially set to `auto`, the interpolation method is interpreted as `nearest` or `linear`, based on whether or not the data is entirely of `integer` type, respectively.  If set to other than `auto`, then no interpretation is applied (the setting is passed through to `np.quantile()` as directed by user).
* Update existing tests and verify their correctness.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-699/GREAT-701
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/core/expect_column_max_to_be_between.py [hunk 1]
```diff
@@ -115,26 +115,25 @@ class ExpectColumnMaxToBeBetween(ColumnExpectation):
...
+        class_name="NumericMetricRangeMultiBatchParameterBuilder",
+        name="max_range_estimator",
+        metric_name="column.max",
+        metric_domain_kwargs=DOMAIN_KWARGS_PARAMETER_FULLY_QUALIFIED_NAME,
+        metric_value_kwargs=None,
+        enforce_numeric_metric=True,
+        replace_nan_with_zero=True,
+        reduce_scalar_metric=True,
+        false_positive_rate=f"{VARIABLES_KEY}false_positive_rate",
+        quantilie_statistic_interpolation_method=f"{VARIABLES_KEY}quantilie_statistic_interpolation_method",
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_max_to_be_between.py]: :heart:

File: great_expectations/expectations/core/expect_column_quantile_values_to_be_between.py [hunk 2]
```diff
@@ -147,29 +147,28 @@ class ExpectColumnQuantileValuesToBeBetween(ColumnExpectation):
...
+        metric_domain_kwargs=DOMAIN_KWARGS_PARAMETER_FULLY_QUALIFIED_NAME,
+        metric_value_kwargs={
+            "quantiles": f"{VARIABLES_KEY}quantiles",
+            "allow_relative_error": f"{VARIABLES_KEY}allow_relative_error",
+        },
+        enforce_numeric_metric=True,
+        replace_nan_with_zero=True,
+        reduce_scalar_metric=True,
+        false_positive_rate=f"{VARIABLES_KEY}false_positive_rate",
+        quantilie_statistic_interpolation_method=f"{VARIABLES_KEY}quantilie_statistic_interpolation_method",
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_quantile_values_to_be_between.py]: I believe this is a typo. Should be `quantile_statistic_interpolation_method`

File: great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py [hunk 3]
```diff
@@ -286,25 +303,67 @@ def _build_parameters(
...
+            round_decimals = 0
+        else:
+            round_decimals = self._get_round_decimals_using_heuristics(
+                metric_values=metric_values,
+                domain=domain,
+                variables=variables,
+                parameters=parameters,
+            )
+
+        if quantilie_statistic_interpolation_method == "auto":
```
Comment (Shinnnyshinshin) [inline great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py]: :heart:

Comment (Shinnnyshinshin) [reply great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py]: and I believe the typo is here too `quantile`

**Review by Shinnnyshinshin — APPROVED** [review]
Looks absolutely wonderful... thank you so much for your help with this Alex

File: great_expectations/expectations/core/expect_column_quantile_values_to_be_between.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/core/expect_column_quantile_values_to_be_between.py]: Fixed -- great catch, thanks @Shinnnyshinshin !

**Pull request merged**